reliability/single-fit/rum/src/test/kotlin/com/datadog/android/rum/integration/tests/assertj/RumEventAssert.kt

```diff
@@ -121,11 +122,25 @@ class RumEventAssert(actual: JsonObject) :
         return this
     }
 
+    fun hasWasStopped(wasStopped: Boolean): RumEventAssert {
+        hasField("additionalProperties.rse.was_stopped", wasStopped)
+        return this
+    }
+
+    fun hasViewCount(viewCount: Int): RumEventAssert {
+        hasField("additionalProperties.rse.views_count.total", viewCount)
+        return this
+    }
+
     // endregion
 
     companion object {
         fun assertThat(actual: JsonObject): RumEventAssert {
             return RumEventAssert(actual)
         }
+
+        fun assertThat(actual: Any): RumEventAssert {
+            return assertThat(Gson().toJsonTree(actual).asJsonObject)
+        }
```

Comment:
IMO this looks like a hack.

Ideally, I don't think it should be `Any` argument type here, and probably this shouldn't be inside `RumEventAssert`. `RumEventAssert` is used for the RUM events only (the ones which are going to be sent to the intake), which are RUM View, RUM Error, etc. events.

In your case you have argument which is not a RUM event (and obviously none of the RUM events can have "wasStopped" or "hasViewCount" attribute). This approach works, but it is mixing the things. Maybe it is better rather to create another assertion class (say for `Map<String, Any>` or even `Map<*,*>`) and since you are not operating with JSON you can write something like that there (let's assume `actual` is of type `Map<String, Any>)`:

```
fun hasWasStopped(wasStopped: Boolean): TelemetryMetricAssert {
        assertThat(actual).containsPathEntry(path, value)
        return this
    }
```

and then you just need to implement `containsPathEntry` which will take the path like `additionalProperties.rse.views_count.total` in a similar manner how it was done for json (just split in the fragments and iterate over).